src/pybind/mgr/prometheus/module.py

```diff
@@ -53,6 +53,12 @@ def health_status_to_number(status):
 DF_POOL = ['max_avail', 'bytes_used', 'raw_bytes_used', 'objects', 'dirty',
            'quota_bytes', 'quota_objects', 'rd', 'rd_bytes', 'wr', 'wr_bytes']
 
+FS_METADATA = ('data_pools', 'id', 'metadata_pool', 'name')
+
+MDS_METADATA = ('id', 'fs', 'public_addr', 'rank')
+
+MON_METADATA = ('id', 'public_addr', 'rank')
```

Comment:
It would be great if we got the hostname, here so that we could match the queries with node_exporter. I poked around the structures exported to the python modules but did not see it there. However, there seems to be some mechanism for this in the C++ ceph-mgr code. At least it looked like that is what we used for osd hostname resolution.

@jcsp Would it be doable to export mon, mds (and ideally rgw, too) hostnames to the python code?

I guess we could try to get hostname from the public_addr in the python code but based on my experience with calamari that is a fairly non-trivial code to implement properly and I suppose it would be better if the hostname resolution code would be aligned with what ceph actually uses.

Reply:
OK, I poked around a bit more and I could get this information from python with `self._ceph_get_server(None)`. It lists all the servers with their hostnames alongside the ceph version and the services that are running on them.

Maybe, we could design a ceph_server_metadata metric that would expose type(s), version and hostname for all the ceph servers?

Reply:
I implemented it in https://github.com/ceph/ceph/pull/20582. It should help us write the queries in an easier way and get the list of osd/mon/rgw/mds hostnames/versions easily.